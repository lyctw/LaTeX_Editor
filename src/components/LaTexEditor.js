import React from 'react';
import MathQuill, { addStyles as addMathquillStyles } from 'react-mathquill'
 
addMathquillStyles()



export default class LaTexEditor extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      latex: ''
    }

    this.mathQuillEl = null

    // this.resetField = () => {
    //   this.mathQuillEl.latex(initialLatex)
    // }
  }

  render() {
    return (
      <div>
        {/* <p style={{color: "cadetblue"}}>DEMO 1: 和word輸入方程式相同，輸入LaTeX語法按空白鍵會自動渲染，並可以在編輯其中的變數，使用範例如下</p>
        <img src={"https://cloud.githubusercontent.com/assets/225809/15163580/1bc048c4-16be-11e6-98a6-de467d00cff1.gif"} width="150px"/><br/> */}
        Math field:{' '}
        <MathQuill
          className="mathquill-field"
          latex={this.state.latex}
          onChange={mathField => {
            const latex = mathField.latex()
            // const text = mathField.text()  // fetch math text
            console.log('latex changed:', latex)
            this.setState({ latex })
          }}
          mathquillDidMount={el => {
            this.mathQuillEl = el
          }}
        />
        <br/>
        <button 
            onClick={() => { 
                const currentTex = this.mathQuillEl.latex();
                this.mathQuillEl.latex('')
                this.props.handleInsert(currentTex);
            }}
        >Insert LaTeX</button> 
      </div>
    )
  }
}