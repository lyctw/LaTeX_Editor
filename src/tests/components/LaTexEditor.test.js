import React from 'react'
import { shallow } from 'enzyme'
import LaTexEditor from '../../components/LaTexEditor'
import toJson from 'enzyme-to-json';

test('should render LaTexEditor', () => {
    const wrapper = shallow(<LaTexEditor />);
    expect(toJson(wrapper)).toMatchSnapshot();
});