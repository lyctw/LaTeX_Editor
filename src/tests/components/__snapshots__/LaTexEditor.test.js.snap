// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LaTexEditor 1`] = `
<div>
  Input LaTeX: 
  <t
    className="mathquill-field"
    latex=""
    mathquillDidMount={[Function]}
    onChange={[Function]}
    style={
      Object {
        "width": "125px",
      }
    }
  />
  <br />
  <br />
  <Button
    as="button"
    buttonRef={[Function]}
    cursor="pointer"
    fluidWidth={false}
    margin="0"
    onClick={[Function]}
    size="medium"
    type="button"
    variant="default"
  >
    Add
  </Button>
</div>
`;
